mmv1/third_party/terraform/resources/resource_container_cluster.go.erb

```diff
@@ -1025,6 +1025,40 @@ func resourceContainerCluster() *schema.Resource {
 				},
 			},
 
+<% unless version == 'ga' -%>
+			"protect_config": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Computed:    true,
+				MaxItems: 	 1,
+				Description: `The notification config for sending cluster upgrade notifications`,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"workload_config": {
+							Type:        schema.TypeList,
+							Optional:    true,
+							MaxItems:    1,
+							Description: `WorkloadConfig defines the flags to enable or disable the workload configurations for the cluster.`,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"audit_mode": {
+										Type:        schema.TypeString,
+										Optional:    true,
+										Description: `Mode defines how to audit the workload configs. Accepted values are MODE_UNSPECIFIED, DISABLED, BASIC.`,
+									},
+								},
+							},
+						},
+						"workload_vulnerability_mode": {
+										Type:             schema.TypeString,
+										Optional:         true,
+										Description: `WorkloadVulnerabilityMode defines mode to perform vulnerability scanning. Accepted values are WORKLOAD_VULNERABILITY_MODE_UNSPECIFIED, DISABLED, BASIC.`,
+									},
```

Comment:
Can we make one of the subfields required (or have AtLeastOneOf added to them) to prevent users from sending empty blocks.

Reply:
made workload_config and audit_mode required to ensure that empty objects are never passed as input. Also added computed:true to workload_vulnerability_mode as default value is added to the field by server if no value is specified.